### PR TITLE
Dockerize AMC

### PR DESCRIPTION
**Pull request by OmarIthawi**
This pull request is part of a group of other PRs to dockerize AMC and revamp the devstack in general in the hope of it being more stable and works better across the platforms.

Generally this is to clean up our mess.

### Related Pull Requests

 - **Devstack:** https://github.com/appsembler/devstack/pull/21
 - **AMC:** https://github.com/appsembler/amc/pull/185

### Trello Card

 [(8) Dockerize AMC in devstack](https://trello.com/c/1BfkvPEn/4094-8-dockerize-amc-in-devstack)

### CODE REVIEW

**Comment by OmarIthawi** [comment]
[(8) Dockerize AMC in devstack](https://trello.com/c/1BfkvPEn/4094-8-dockerize-amc-in-devstack)

**Comment by thraxil** [comment]
Currently, I'm getting this:

```
Recreating tahoe.devstack.postgres ... done
Creating tahoe.devstack.amc        ... done
make[1]: Leaving directory '/home/anders/work/tahoe-devstack/devstack'
make tahoe.provision
make[1]: Entering directory '/home/anders/work/tahoe-devstack/devstack'
make tahoe.chown
make[2]: Entering directory '/home/anders/work/tahoe-devstack/devstack'
sudo chown anders -R /home/anders/work/tahoe-devstack/devstack/..
[sudo] password for anders: 
make[2]: Leaving directory '/home/anders/work/tahoe-devstack/devstack'
cat /home/anders/work/tahoe-devstack/devstack/../devstack/provision-tahoe.py > /home/anders/work/tahoe-devstack/devstack/../src/provision-tahoe.py
make COMMAND='python /edx/src/provision-tahoe.py' tahoe.exec.edxapp
make[2]: Entering directory '/home/anders/work/tahoe-devstack/devstack'
make COMMAND='python /edx/src/provision-tahoe.py' SERVICE=lms tahoe.exec.single
make[3]: Entering directory '/home/anders/work/tahoe-devstack/devstack'
docker exec -t edx.devstack.lms  \
	bash -c 'source /edx/app/edxapp/edxapp_env && cd /edx/app/edxapp/edx-platform/ && python /edx/src/provision-tahoe.py'
make[3]: Leaving directory '/home/anders/work/tahoe-devstack/devstack'
make COMMAND='python /edx/src/provision-tahoe.py' SERVICE=studio tahoe.exec.single
make[3]: Entering directory '/home/anders/work/tahoe-devstack/devstack'
docker exec -t edx.devstack.studio  \
	bash -c 'source /edx/app/edxapp/edxapp_env && cd /edx/app/edxapp/edx-platform/ && python /edx/src/provision-tahoe.py'
make[3]: Leaving directory '/home/anders/work/tahoe-devstack/devstack'
make[2]: Leaving directory '/home/anders/work/tahoe-devstack/devstack'
rm /home/anders/work/tahoe-devstack/devstack/../src/provision-tahoe.py
make tahoe.restart || true
make[2]: Entering directory '/home/anders/work/tahoe-devstack/devstack'
make lms-restart
make[3]: Entering directory '/home/anders/work/tahoe-devstack/devstack'
docker exec -t edx.devstack.lms bash -c 'kill $(ps aux | grep "manage.py lms" | egrep -v "while|grep" | awk "{print \$2}")'
make[3]: Leaving directory '/home/anders/work/tahoe-devstack/devstack'
make studio-restart
make[3]: Entering directory '/home/anders/work/tahoe-devstack/devstack'
docker exec -t edx.devstack.studio bash -c 'kill $(ps aux | grep "manage.py cms" | egrep -v "while|grep" | awk "{print \$2}")'
make[3]: Leaving directory '/home/anders/work/tahoe-devstack/devstack'
make amc-restart
make[3]: Entering directory '/home/anders/work/tahoe-devstack/devstack'
docker exec -t tahoe.devstack.amc bash -c 'killall5'
docker exec -t tahoe.devstack.amc-frontend bash -c 'killall5'
make[3]: Leaving directory '/home/anders/work/tahoe-devstack/devstack'
make[2]: Leaving directory '/home/anders/work/tahoe-devstack/devstack'
make[1]: Leaving directory '/home/anders/work/tahoe-devstack/devstack'
make amc.provision
make[1]: Entering directory '/home/anders/work/tahoe-devstack/devstack'
docker exec -it tahoe.devstack.amc python manage.py migrate
python: can't open file 'manage.py': [Errno 2] No such file or directory
tahoe.mk:36: recipe for target 'amc.provision' failed
make[1]: *** [amc.provision] Error 2
make[1]: Leaving directory '/home/anders/work/tahoe-devstack/devstack'
Makefile:62: recipe for target 'dev.provision' failed
make: *** [dev.provision] Error 2
```

but I suspect that it is because of the issue I had with https://github.com/appsembler/amc/pull/185

**Comment by OmarIthawi** [comment]
Yes @thraxil! Just pull the images and you should be ok:

```bash
$ docker pull gcr.io/appsembler-tahoe-0/amc-frontend
$ docker pull gcr.io/appsembler-tahoe-0/amc
```

File: tahoe.mk [hunk 1]
```diff
@@ -24,131 +8,42 @@ tahoe.exec.edxapp:   ## Execute a command in both LMS and Studio (edxapp contain
...
-	sudo rm -rf $(DEVSTACK_WORKSPACE)/src/themes/customer_themes
-	mkdir -p $(DEVSTACK_WORKSPACE)/src/themes/customer_themes/  # Make it writeble for the user
-
-	git clone git@github.com:appsembler/edx-theme-codebase.git $(THEMES_DIR)/edx-theme-codebase
-	cd $(THEMES_DIR)/edx-theme-codebase && git checkout hawthorn/master
-
-	git clone git@github.com:appsembler/edx-theme-customers.git $(CUSTOMER_THEME_DIR)
-	cd $(CUSTOMER_THEME_DIR) && git checkout hawthorn/tahoe
+tahoe.chown:  ## Fix annoying docker permission issues
+	sudo chown $(USER) -R $(DEVSTACK_WORKSPACE)
```
Comment (melvinsoft) [inline tahoe.mk]:
```suggestion
	sudo chown -R $(USER) $(DEVSTACK_WORKSPACE)
```

**Review by melvinsoft — CHANGES_REQUESTED** [review]
Just a minor change for now.

File: tahoe.mk (hunk 1 above)
Comment (OmarIthawi) [reply tahoe.mk]: Why @melvinsoft? Doesn't it work on Mac?

**Review by thraxil — APPROVED** [review]
Basically working for me. I'm running into JS errors on signup, but that's probably separate.

**Comment by OmarIthawi** [comment]
Thanks @thraxil! I think that's an error in the devstack that appears as a JavaScript error.

I've pushed a fix to add the directory `customer_themes` so the trial signup can work.

Please let me know if it works for you now.

**Pull request merged**